src/transformers/models/superglue/modeling_superglue.py

```diff
@@ -0,0 +1,845 @@
+from transformers.models.superglue.configuration_superglue import SuperGlueConfig
+
+from ...utils import (
+    ModelOutput,
+    add_start_docstrings_to_model_forward,
+    logging,
+)
+from ..auto import AutoModelForKeypointDetection
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC_ = "SuperGlueConfig"
+
+_CHECKPOINT_FOR_DOC_ = "stevenbucaille/superglue_indoor"
+
+
+def concat_attentions_tuples_pair(
+    attention_probs_0: Tuple[torch.Tensor], attention_probs_1: Tuple[torch.Tensor]
+) -> Tuple[torch.Tensor]:
+    """
+    Concatenate two tuple of attention probabilities into one.
+    We assume that the attention probabilities are of shape (batch_size, num_heads, num_keypoints_a, num_keypoints_b).
+    The tuples are assumed to have the same length and of the following form:
+    attention_probs_0 = (attention_probs_0_0, attention_probs_0_1, ...)
+    attention_probs_1 = (attention_probs_1_0, attention_probs_1_1, ...)
+    The output will then be of the form:
+    new_attention_probs = (torch.cat(attention_probs_0_0, attention_probs_1_0), torch.cat(attention_probs_0_1, attention_probs_1_1), ...)
+    If the attention probabilities have different shapes, the smaller one will be padded with zeros :
+    (batch_size * 2, num_heads, max(num_keypoints_a_0, num_keypoints_a_1), max(num_keypoints_b_0, num_keypoints_b_1))
+    """
+    new_attention_probs = ()
+    for attention_prob_0, attention_prob_1 in zip(attention_probs_0, attention_probs_1):
+        if attention_prob_0.size() != attention_prob_1.size():
+            max_dim2 = max(attention_prob_0.shape[2], attention_prob_1.shape[2])
+            max_dim3 = max(attention_prob_0.shape[2], attention_prob_1.shape[2])
+            new_attention_prob = torch.zeros(2, attention_prob_0.shape[1], max_dim2, max_dim3).to(
+                attention_prob_0.device
+            )
+            new_attention_prob[0, :, : attention_prob_0.shape[2], : attention_prob_0.shape[3]] = attention_prob_0
+            new_attention_prob[1, :, : attention_prob_1.shape[2], : attention_prob_1.shape[3]] = attention_prob_1
+            new_attention_probs = new_attention_probs + (new_attention_prob,)
+        else:
+            new_attention_probs = new_attention_probs + (torch.cat([attention_prob_0, attention_prob_1]),)
+    return new_attention_probs
+
+
+def stack_attention_probs_list(attention_probs: List[torch.Tensor]) -> torch.Tensor:
+    all_attention_probs_shape_the_same = True
+    first_attention_prob_shape = attention_probs[0].shape
+    for attention_prob in attention_probs[1:]:
+        if attention_prob.shape != first_attention_prob_shape:
+            all_attention_probs_shape_the_same = False
+    if all_attention_probs_shape_the_same:
+        stacked_attention_probs = torch.stack(attention_probs, dim=0)
+    else:
+        max_dim2 = max([attention_prob.shape[2] for attention_prob in attention_probs])
+        max_dim3 = max([attention_prob.shape[3] for attention_prob in attention_probs])
```

Comment:
You don't need to create a list 
```suggestion
        max_dim2 = max(attention_prob.shape[2] for attention_prob in attention_probs)
        max_dim3 = max(attention_prob.shape[3] for attention_prob in attention_probs)
```

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/29886/commits/601da1cf0756cef7339d4dd33faeefbfcd073ac9)